### PR TITLE
Added ESXi Support into 1 commit

### PR DESCRIPTION
**Pull request by redwiz666**

### CODE REVIEW

**Comment by redwiz666** [comment]
@cliffe Here is all the ESXi support in one commit. Let me know if you have any questions and I can explain it.
I put a link to the provider plugin that I used below
https://github.com/josenk/vagrant-vmware-esxi

File: lib/schemas/base_metadata_schema.xsd [hunk 1]
```diff
@@ -50,6 +50,7 @@
         <xs:element name="platform" type="platformOptions" minOccurs="1" maxOccurs="unbounded"/>
         <xs:element name="distro" type="xs:string" minOccurs="1" maxOccurs="1"/>
         <xs:element name="url" type="xs:string" minOccurs="1" maxOccurs="1"/>
+		<xs:element name="esxi_url" type="xs:string" minOccurs="0" maxOccurs="1"/>
```
Comment (cliffe) [inline lib/schemas/base_metadata_schema.xsd]: Spaces rather than tabs please :)

File: lib/templates/Vagrantfile.erb [hunk 2]
```diff
@@ -18,6 +18,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       if selected_module.module_type == 'base'
         @base_type = selected_module.attributes['type']
         @ovirt_template = selected_module.attributes['ovirt_template']
+	@esxi_url = selected_module.attributes['esxi_url']
```
Comment (cliffe) [inline lib/templates/Vagrantfile.erb]: Spaces

File: secgen.rb [hunk 3]
```diff
@@ -182,6 +191,15 @@ def build_vms(scenario, project_dir, options)
   end
   if successful_creation
     ovirt_post_build(options, scenario, project_dir) if OVirtFunctions.provider_ovirt?(options)
+        if options[:snapshot]
```
Comment (cliffe) [inline secgen.rb]: Indenting

**Pull request merged**